torch/_inductor/select_algorithm.py

```diff
@@ -649,6 +666,7 @@ def make_kernel_render(out_node):
                 "allow_tf32": str(kwargs.get("ALLOW_TF32", None)),
                 "acc_type": str(kwargs.get("ACC_TYPE", None)),
             },
+            mutated_inputs=mutated_inputs,
```

Comment:
Could we add a test that we don't fuse any use of the mutated input into the templated attention ?

Could we also check that we are correctly cloning the input mutated args here in benchmark ?

Reply:
To be clear, no other kernels are actually being fused into templated attention. And the input buffer that's being mutated is only being created for this kernel specifically.

We're only marking it as "mutated" because triton templates (and in general, Triton kernels) do not directly support multiple outputs that are being allocated. So instead,  we say it has "one output, and two 'inputs' that are mutated". 

@drisspg This would be a good thing to make clear as a comment.

Reply:
I left a note here: https://github.com/pytorch/pytorch/pull/124308/files#diff-cf6ca00beddc32a2a6a2933fb9913b6a2b925ffc3b745488967210e4343134acR3646

Do you think I should move it, I also added a see note above but could move it down here

Reply:
You can probably also add a note in the FlashAttention lowering